### ISSUE TITLE
Error: Your Skill Manifest is not compatible. Please note that the minimum supported manifest version is 2.1.

### ISSUE THREAD

**Issue by badhonkundu** [microsoft/botframework-solutions#3271]
#### What project is affected?
Virtual Assistant - While adding Skill to the VA
#### What language is this in?
csharp
#### What happens?
We get the below error while running the command
botskills connect --remoteManifest "http://<YOUR_SKILL_NAME>.azurewebsites.net/manifest/manifest-1.1.json" --cs
ERROR:
There was an error while connecting the Skill to the Assistant:
Error: Your Skill Manifest is not compatible. Please note that the minimum supported manifest version is 2.1.
#### What are the steps to reproduce this issue?
Create and deploy VA to Azure. Create a skill and deploy to azure. (Both following steps in https://microsoft.github.io/botframework-solutions/). Check individual VA and the Skill is working individually.
Try to add the skill to the Virtual Assistant using the below command:
botskills connect --remoteManifest "http://<YOUR_SKILL_NAME>.azurewebsites.net/manifest/manifest-1.1.json" --cs
<YOUR_SKILL_NAME> = App service URL for the azure deployed skill
#### What were you expecting to happen?
The skill to get added to the virtual assistant without any issues
#### Any screenshots or additional context?
![image](https://user-images.githubusercontent.com/63103301/78695828-3f85cd00-791c-11ea-9e41-5af9b773cb0e.png)

**Comment by VictorGrycuk** [comment]
Hi @badhonkundu!

We will take a look at this and get back to you.

**Comment by pavolum** [comment]
@badhonkundu can you send over a link to your 1.1 manifest or a screenshot/copy? I remember running into this issue in the past and I believe it was when we moved to the new manifest schema. If you are using an older skill then the manifest version/schema may be mismatched. 

I have just run a fresh VA and a fresh Skill and the connected just fine using the the most up to date botskills CLI, VA and Skills template.

For reference here is the 1.1 skill manifest that I just used, please check and make sure the schema and version number between this one and yours aligns  
```
{
  "$schema": "https://schemas.botframework.com/schemas/skills/skill-manifest-2.1.preview-0.json",
  "$id": "TestSkillDelete",
  "name": "TestSkillDelete",
  "description": "TestSkillDelete description",
  "publisherName": "Your Company",
  "version": "1.1",
  "iconUrl": "https://{YOUR_SKILL_URL}/TestSkillDelete.png",
  "copyright": "Copyright (c) Microsoft Corporation. All rights reserved.",
  "license": "",
  "privacyUrl": "https://{YOUR_SKILL_URL}/privacy.html",
  "tags": [
    "sample",
    "skill"
  ],
  "endpoints": [
    {
      "name": "production",
      "protocol": "BotFrameworkV3",
      "description": "Production endpoint for the TestSkillDelete",
      "endpointUrl": "https://{YOUR_SKILL_URL}/api/messages",
      "msAppId": "{YOUR_SKILL_APPID}"
    }
  ],
  "dispatchModels": {
    "languages": {
      "en-us": [
        {
          "id": "TestSkillDeleteLuModel-en",
          "name": "TestSkillDelete LU (English)",
          "contentType": "application/lu",
          "url": "file://TestSkillDelete.lu",
          "description": "English language model for the skill"
        }
      ],
      "de-de": [
        {
          "id": "TestSkillDeleteLuModel-de",
          "name": "TestSkillDelete LU (German)",
          "contentType": "application/lu",
          "url": "file://TestSkillDelete.lu",
          "description": "German language model for the skill"
        }
      ],
      "es-es": [
        {
          "id": "TestSkillDeleteLuModel-es",
          "name": "TestSkillDelete LU (Spanish)",
          "contentType": "application/lu",
          "url": "file://TestSkillDelete.lu",
          "description": "Spanish language model for the skill"
        }
      ],
      "fr-fr": [
        {
          "id": "TestSkillDeleteLuModel-fr",
          "name": "TestSkillDelete LU (French)",
          "contentType": "application/lu",
          "url": "file://TestSkillDelete.lu",
          "description": "French language model for the skill"
        }
      ],
      "it-it": [
        {
          "id": "TestSkillDeleteLuModel-it",
          "name": "TestSkillDelete LU (Italian)",
          "contentType": "application/lu",
          "url": "file://TestSkillDelete.lu",
          "description": "Italian language model for the skill"
        }
      ],
      "zh-cn": [
        {
          "id": "TestSkillDeleteLuModel-zh",
          "name": "TestSkillDelete LU (Chinese)",
          "contentType": "application/lu",
          "url": "file://TestSkillDelete.lu",
          "description": "Chinese language model for the skill"
        }
      ]
    },
    "intents": {
      "Sample": "#/activities/message",
      "*": "#/activities/message"
    }
  },
  "activities": {
    "sampleAction": {
      "description": "Sample action which accepts an input object and returns an object back.",
      "type": "event",
      "name": "SampleAction",
      "value": {
        "$ref": "#/definitions/inputObject"
      },
      "resultValue": {
        "$ref": "#/definitions/responseObject"
      }
    },
    "message": {
      "type": "message",
      "description": "Receives the users utterance and attempts to resolve it using the skill's LU models"
    }
  },
  "definitions": {
    "inputObject": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string",
          "description": "The users name."
        }
      }
    },
    "responseObject": {
      "type": "object",
      "properties": {
        "customerId": {
          "type": "integer",
          "description": "A customer identifier."
        }
      }
    }
  }
}
```

**Comment by badhonkundu** [comment]
@pavolum Below is the schema from my solution (I have removed the appIDs and urls)
I am using a fresh Skill and latest botskills CLI too.
{
  "$schema": "https://schemas.botframework.com/schemas/skills/skill-manifest-2.1.preview-0.json",
  "$id": "Skill_HR",
  "name": "Skill_HR",
  "description": "Skill_HR description",
  "publisherName": "Your Company",
  "version": "2.1",
  "iconUrl": "https://{YOUR_SKILL_URL}.azurewebsites.net/Skill_HR.png",
  "copyright": "Copyright (c) Microsoft Corporation. All rights reserved.",
  "license": "",
  "privacyUrl": "https://{YOUR_SKILL_URL}/privacy.html",
  "tags": [
    "sample",
    "skill"
  ],
  "endpoints": [
    {
      "name": "production",
      "protocol": "BotFrameworkV3",
      "description": "Production endpoint for the Skill_HR",
      "endpointurl": "https://{your_skill_url}/api/messages",
      "msAppId": "{YOUR_SKILL_APPID}"
    }
  ],
  "dispatchModels": {
    "languages": {
      "en-us": [
        {
          "id": "Skill_HRLuModel-en",
          "name": "Skill_HR LU (English)",
          "contentType": "application/lu",
          "url": "file://Skill_HR.lu",
          "description": "English language model for the skill"
        }
      ],
      "de-de": [
        {
          "id": "Skill_HRLuModel-de",
          "name": "Skill_HR LU (German)",
          "contentType": "application/lu",
          "url": "file://Skill_HR.lu",
          "description": "German language model for the skill"
        }
      ],
      "es-es": [
        {
          "id": "Skill_HRLuModel-es",
          "name": "Skill_HR LU (Spanish)",
          "contentType": "application/lu",
          "url": "file://Skill_HR.lu",
          "description": "Spanish language model for the skill"
        }
      ],
      "fr-fr": [
        {
          "id": "Skill_HRLuModel-fr",
          "name": "Skill_HR LU (French)",
          "contentType": "application/lu",
          "url": "file://Skill_HR.lu",
          "description": "French language model for the skill"
        }
      ],
      "it-it": [
        {
          "id": "Skill_HRLuModel-it",
          "name": "Skill_HR LU (Italian)",
          "contentType": "application/lu",
          "url": "file://Skill_HR.lu",
          "description": "Italian language model for the skill"
        }
      ],
      "zh-cn": [
        {
          "id": "Skill_HRLuModel-zh",
          "name": "Skill_HR LU (Chinese)",
          "contentType": "application/lu",
          "url": "file://Skill_HR.lu",
          "description": "Chinese language model for the skill"
        }
      ]
    },
    "intents": {
      "Sample": "#/activities/message",
      "*": "#/activities/message"
    }
  },
  "activities": {
    "sampleAction": {
      "description": "Sample action which accepts an input object and returns an object back.",
      "type": "event",
      "name": "SampleAction",
      "value": {
        "$ref": "#/definitions/inputObject"
      },
      "resultValue": {
        "$ref": "#/definitions/responseObject"
      }
    },
    "message": {
      "type": "message",
      "description": "Receives the users utterance and attempts to resolve it using the skill's LU models"
    }
  },
  "definitions": {
    "inputObject": {
      "type": "object",
      "properties": {
        "name": {
          "type": "string",
          "description": "The users name."
        }
      }
    },
    "responseObject": {
      "type": "object",
      "properties": {
        "customerId": {
          "type": "integer",
          "description": "A customer identifier."
        }
      }
    }
  }
}

**Comment by pavolum** [comment]
@badhonkundu please change the version version attribute from 2.2 to 1.1 and see if that resolves your issue

**Comment by badhonkundu** [comment]
@pavolum 

It was originally "version": "1.1" go the same error.

I was trying my luck to see if it helps if we change the version number to 2.1. Same result.

I checked now again changing it to 1.1  which resulted in the same error.

**Comment by pavolum** [comment]
@badhonkundu may be a casing issue oddly enough...
In this portion of your manifest
```
"endpoints": [
{
"name": "production",
"protocol": "BotFrameworkV3",
"description": "Production endpoint for the Skill_HR",
"endpointurl": "https://{your_skill_url}/api/messages",
"msAppId": "{YOUR_SKILL_APPID}"
}
```

Change "endpointurl" to "endpointUrl". I was able to reproduce your version error when I changed this to what you had ("endpointurl"), after reverting back to the camel case version ("endpointUrl") I was able to connect.

**Comment by badhonkundu** [comment]
It is "endpointUrl" in the manifest. I must have messed it up when replacing the original URL to share the manifest here. My bad! Sincere apologies about that. Just to ensure, I tried it again with no luck.

**Comment by pavolum** [comment]
@badhonkundu damn I thought that may have been the issue since the change did result in the same error message.

Can you ensure that your published manifest matches the one I sent earlier in this thread as far as schema, versions and attribute names/casing. Also can you send over the version of botskills you are running?

@Batta32 any thoughts?

**Comment by ipraveenMS** [comment]
I see the same exact issue. endpointUrl is correct in my manifest. Pls help as its blocking issue

I provisioned in a new machine using steps https://microsoft.github.io/botframework-solutions/virtual-assistant/tutorials/create-assistant/csharp/1-intro/  and https://microsoft.github.io/botframework-solutions/skills/tutorials/create-skill/csharp/7-add-your-skill/ 

botskills -v
1.0.14

**Comment by ipraveenMS** [comment]
For me this is resolved by updating MANUALLY the manifest version locally and then copying it to azure svc manually. I am hitting another error but its past this issue.

**Comment by badhonkundu** [comment]
@ipraveenMS can you confirm some details please?
Did you update the "version" key to "2.1" in the "manifest-1-1.json" file and then manually copy the "manifest-1-1.json" to the azure svc?

**Comment by badhonkundu** [comment]
@pavolum This particular issue got resolved after reforming the manifest-1.1 file. Also noticed if we have commented lines in the manifest, we get that same error.

Now, it throws ANOTHER ERROR:
![image](https://user-images.githubusercontent.com/63103301/78989028-12b9fb80-7b50-11ea-9691-1535633ea295.png)


And I am using botskills@1.0.14

**Comment by pavolum** [comment]
@badhonkundu this additional error you are having is easily resolved, add the --luisFolder flag followed by the path to the LU directory for your skill. For example...
```
--luisFolder "C:/Users/pavolum/source/repos/TestSkillDelete/TestSkillDelete/Deployment/Resources/LU/"
```

**Comment by badhonkundu** [comment]
@pavolum Thanks, that helped resolve the issue. I faced couple of other issues after that, which I was able to fix and at last the skill has been successfully added to my VA.

I will put in few pointers that may help others coming to this thread:

- Install all packages as described in [this link](https://microsoft.github.io/botframework-solutions/skills/handbook/add-skills-to-a-virtual-assistant/), if you haven't already.
- DO NOT keep commented codes/lines in your manifest files.
